### ISSUE TITLE
Wrong HTTP method used on a supported path doesn't produce any log message

### ISSUE THREAD

**Issue by jwadolowski** [Cognifide/knotx#344]
## Steps to reproduce

* deploy knot.x extension that handles `POST /services/search/products/details` requests
* make sure routing is configured accordingly (take a look at config snippet below)
* send `GET` request to `/services/search/products/details`

## Expected state

* 404 response & information in the logs about unsupported path

## Current state

* 404 gets back to the client
* Lack of any message in the logs

## Noteworthy factoids

* call to a path that's **not** listed in the configuration anywhere causes expected log message: `WARN  i.k.s.SupportedMethodsAndPathsHandler - Requested path /test is not supported based on configuration`
* log level was increased to `TRACE`, but that didn't help
* configuration snippet that may be useful

```javascript
"customFlow": {
    "...",
    "routing": {
        "GET": [
            "..."
        ],
        "POST": [
            {
                "address": "knotx.gateway.gatewayknot",
                "onTransition": {
                    "next": {
                        "address": "search.products.details"
                    }
                },
                "path": "/services/search/products/details"
            },
            "..."
        ]
    }
}
```

**Comment by marcinczeczko** [comment]
I see that you're actually getting the WARN message in LOG about that situation, whilest in "current state" you pointed out that no appropriate logs.
Could you please clarify what's your expectations?

**Comment by jwadolowski** [comment]
The problem occurs when you use `GET` method on a supported path (defined in routing) that accepts only `POST` requests, i.e `POST /my/endpoint` is supported, but I sent `GET /my/endpoint` instead.

Proper log message gets printed when you ask for a path that's not present in routing configuration, i.e. `/test` in my example above.

**Comment by marcinczeczko** [comment]
Fixed in #347
